### ISSUE TITLE
Questions on finetuning new data

### ISSUE THREAD

**Issue by willtsai060** [microsoft/BiomedParse#54]
Dear author,
First of all, thanks for the great work!

I have a project about detect DCIS(Ductal Carcinoma In Situ) and IDC(Invasive Ductal Carcinoma) in breast cancer pathology H&E image and would like to use your model to fine-tune my data.  My first goal is to get a tumor region-level segmetation rather than cell-level.

Currently, I’ve tried training on IDC data alone. In `label_base.json`, I found similar labels like `"10": "tumor"` and `"15": "histology"`. My approach was rename the IDC masks to `[IMAGE-NAME]_pathology_breast_tumor.png` and proceed with training. I then followed the instructions in the README to create the dataset. This method allowed me to successfully train the model, and the results seemed to meet my expectations. However, I’m not sure if this approach is correct, or if I should add an "IDC" category as a child of "tumor" and rename the masks to "IDC" instead.

Another issue I encountered was when I tried fine-tuning both DCIS and IDC simultaneously. The method I used was relabel `"10": "tumor"` to "IDC" and `"15": "histology"` to "DCIS". I named the masks as `[IMAGE-NAME]_pathology_breast_IDC.png` and `[IMAGE-NAME]_pathology_breast_DCIS.png`
After fine-tuning, during inference, I used the prompt ['IDC', 'DCIS'] to get the output. However, the resulting two masks looked almost identical, with only slight differences in small details.

To summarize my questions:

1. What is the best practice for fine-tuning labels that are not in the original model?
2. When fine-tuning a specific category, such as "tumor", will other labels (like "background", etc.) be affected? I’ve observed that after fine-tuning "tumor," the output segmentation for other categories also changed.
3. How should prompts be used? For example, if I use the prompt ["neoplastic"], which category does it actually correspond to? I found these labels in `label_base.json`: "**neoplastic polyp**", "**non-neoplastic polyp**" in "lesion",  "**neoplastic cells**" in "histology structure". Or in my case, is using ['IDC', 'DCIS'] correct?

Looking forward to your reply. Any advice would be greatly appreciated.

**Comment by theodore-zhao** [comment]
Thanks for your interest in our work! Glad that the finetuning is working to some extent. To your questions:

1. The class label is not the key in differentiating the segmentation objectives. The text prompt is actually related to the desired segmentation target. Therefore, it is okay to have the new object type under any existing meta object type that is most relevant. Otherwise, it is okay to just put in the "other" category.
2. The segmentation output change that you observed after finetuning is due to the model weight change. After finetuning exclusively on certain data, the model will pick up on those tasks but the performance on other pretrained tasks will decrease as they use shares model weights. If you want to keep performance on certain pretrained tasks at similar level, it is recommended that you use the corresponding preprocess datasets that we released in the finetuning as well. If you have multiple tasks for finetuning, it is okay to just train them at the same time.
3. For your case, I think the best prompt would be spelling out the full name. You can specify that they are in breast pathology.

Hope these helps!

**Comment by willtsai060** [comment]
Hi @vusynhathoang,  I'm using a 24gb 3090ti for finetuning, with batch size 1, which requires approximately 16GB.
I've also tested on a 12gb 4070ti, it also works but with a relatively slow speed. It seems that the program handles insufficient VRAM somewhere?
